### PR TITLE
[onert] Lowering multiple model with controlflow

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit enables lowering subgraphs for multiple model with controlflow. 
Executor generation and execution is not supported yet.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: https://github.com/Samsung/ONE/issues/9713
Draft: https://github.com/Samsung/ONE/pull/9716

### CODE REVIEW

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 1]
```diff
@@ -516,78 +516,66 @@ std::shared_ptr<CompilerArtifact> Compiler::compile(void)
...
-      // For multimodel, model index is used for lowered graph index in lowered graph map
-      // and index type is SubgraphIndex
-      // TODO Find better way to represent lowered graph index for multimodel's subgraph
-      lowered_subgs[ir::SubgraphIndex{i}] =
-        std::make_unique<compiler::LoweredGraph>(*model->primary_subgraph(), *_voptions[i]);
-    }
+  for (uint32_t i = 0; i < model_count; i++)
+  {
+    auto const index_m = ir::ModelIndex{i};
+    auto model = _nnpkg->model(ir::ModelIndex{i});
```
Comment (glistening) [inline runtime/onert/core/src/compiler/Compiler.cc:535]:
```suggestion
    auto model = _nnpkg->model(index_m);
```

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 2]
```diff
@@ -516,78 +516,66 @@ std::shared_ptr<CompilerArtifact> Compiler::compile(void)
...
+
+    model->iterate([&](const ir::SubgraphIndex &index, ir::Graph &subg) {
+      dot_dumper.dump(subg, nnfw::misc::str("before_lower_model-", i, "-subg-", index.value()));
+      // Lower: Assign backend
+      lowered_subgs[index_m][index] = std::make_unique<compiler::LoweredGraph>(subg, *_voptions[0]);
```
[the remark this answers is not in the document]
Comment (hseok-oh) [reply runtime/onert/core/src/compiler/Compiler.cc:540]: It's bug. This line is not updated. You can find this line is not changed if you use `hide whitespace`. I'll update this lline.

[the remark this answers is not in the document]
Comment (glistening) [reply runtime/onert/core/src/compiler/Compiler.cc:540]: @hseok-oh As I understand, `*_voptions[0]` is from `model_count == 1` case. It was `*_voptions[i]` for for loop (multiple model case).

[the remark this answers is not in the document]
Comment (hseok-oh) [reply runtime/onert/core/src/compiler/Compiler.cc]: You're right. I'll update.

**Review by glistening — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**